efi/shim_test.go

```diff
@@ -400,7 +400,7 @@ func (s *shimSuite) TestShimSbatPolicyLatestUnset(c *C) {
 
 	c.Assert(visitor.varModifiers, HasLen, 1)
 
-	collector := NewVariableSetCollector(efitest.NewMockHostEnvironment(nil, nil))
+	collector := NewVariableSetCollector(efitest.NewMockHostEnvironment(efitest.MockVars{}, nil))
```

Comment:
what's the reason for the need of this change and other similar ones?

Reply:
Ah, it was because I made a change to `MockHostEnvironment.VarContext` to not attach a `efi.VarsBackend` if the supplied `efitest.MockVars` map is nil, which resulted in anything that ends up calling `efi.ReadVariable` crashing. I reverted the change to how it worked before - `MockHostEnvironment.VarContext` will always add a `efi.VarsBackend` even if the supplied `efitest.MockVars` is nil. This does mean that anything which calls `efi.ReadVariable` ends up reading keys from a nil map, but that seems to be ok in go (it's only adding entries to a nil map that causes a panic)